hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/blockmanagement/DatanodeManager.java

```diff
@@ -376,10 +373,14 @@ public void initSlowPeerTracker(Configuration conf, Timer timer,
     this.slowPeerTracker = dataNodePeerStatsEnabled ?
         new SlowPeerTracker(conf, timer) :
         new SlowPeerDisabledTracker(conf, timer);
+    if (slowPeerTracker.isSlowPeerTrackerEnabled()) {
+      startSlowPeerCollector();
+    }
```

Comment:
should we stop the running thread if `slowPeerTracker.isSlowPeerTrackerEnabled()` return `false`?

Reply:
thanks @ZanderXu help review it.  
yeah,  you are right, if slowPeerTracker.isSlowPeerTrackerEnabled() is false, better to run stop thread.